tests/test_server.py

```diff
@@ -220,6 +221,9 @@ def test_can_run_main(self, ioloop_mock, get_server_parameters_mock, server_mock
         main()
         ioloop_instance_mock.start.assert_any_call()
 
+    def cleanup(self):
+        ServerTestCase.cleanup_called = True
+
     @mock.patch.object(thumbor.server, 'HTTPServer')
     @mock.patch.object(thumbor.server, 'get_server_parameters')
     @mock.patch('tornado.ioloop.IOLoop.instance', create=True)
```

Comment:
You can use @mock.patch.object on BaseEngine, instead of setting your own method, and then use assert_called.